### PR TITLE
fix: enable symbology attributes for tiled vector layers

### LINKED ISSUE #1475: [Bug]: Getting "no attribute found" when uploading/connecting a vector layer
### Before you start

- [x] I searched the [existing issues](https://github.com/opengeos/GeoLibre/issues) and did not find a duplicate.
- [x] This is a bug report, not a usage question (questions belong in [Discussions](https://github.com/opengeos/GeoLibre/discussions)).

### What happened?

When i upload a vector file and try to adjust the style by category/graduated (or any other), it shows the error of "no attribute found" despite there being many attributes. The specific example I used was a geopackage file. As you can see the tooltip has various attributes - I specifically want to select the "Cluster Name" attribute but the selection states "no attribute found". This also occurs for integer/float data when trying to use graduated style.

### Steps to reproduce

1. Load up a vector layer (I used gepoackage)
2. try to select a different style (graduated/categorized etc)
3. unable to select attribute under selected style.

### Expected behavior

I am able to make an adjustment to the categories/breaks for the various style guide

### How are you running GeoLibre?

web app

### Operating system and browser

Windows 11

### Sample data

_No response_

### Screenshots or logs

_No response_

### PR DESCRIPTION
**Pull request by giswqs**
## Summary

- upgrade `maplibre-gl-vector` to 0.10.2 so tiled layers expose their attribute fields and selected column values
- load only the selected attribute for categorized and graduated styling without materializing tiled geometry
- handle large numeric columns without exceeding JavaScript function argument limits
- move classification helpers into a testable module and add regression coverage

Upstream support: opengeos/maplibre-gl-vector#47
Release: https://github.com/opengeos/maplibre-gl-vector/releases/tag/v0.10.2

Fixes #1475

## Testing

- `pre-commit run --all-files`
- frontend test suite with coverage
- worker type checks
- Rust check
- Playwright verification with a 60,000-feature GeoPackage in light categorized and dark graduated modes

## Summary by CodeRabbit

* **New Features**
  * Added categorized and graduated vector styling driven by attribute values in tiled vector layers (including when GeoJSON data isn’t in memory).
  * Introduced async attribute-value loading with clear UI feedback and safer “apply styling” behavior.
* **Bug Fixes**
  * Avoided unnecessary stop regeneration by reusing already-loaded attribute values.
  * Prevented applying vector styling while attribute values are unavailable.
* **Tests**
  * Added automated tests for categorized and graduated stop generation edge cases.
* **Chores**
  * Updated the vector rendering dependency.

### CODE REVIEW

File: apps/geolibre-desktop/src/components/panels/StylePanel.tsx [hunk 1]
```diff
@@ -1224,6 +1150,96 @@ export function StylePanel({
...
+  }, [
+    draftVectorStyleClassCount,
+    draftVectorStyleClassificationScheme,
+    draftVectorStyleColorRamp,
+    draftVectorStyleMode,
+    draftVectorStyleProperty,
+    layer,
+    loadedVectorPropertyValues,
+  ]);
+
```
[the remark this answers is not in the document]
Comment (giswqs) [reply apps/geolibre-desktop/src/components/panels/StylePanel.tsx:1250]: Fixed in 5fcc1a7c. The regeneration effect now depends only on the relevant layer fields and passes a classification-only layer shape, so unrelated style edits do not reset draft stops.

File: apps/geolibre-desktop/src/lib/vector-style-classification.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+  layer: ClassifiableLayer,
+  property: string,
+  classCount: number,
+  colorRamp: string,
+  classificationScheme: string,
+  propertyValues?: unknown[],
+): VectorStyleStop[] {
+  const values = (propertyValues ?? getPropertyValues(layer, property))
+    .map((value) => Number(value))
+    .filter(Number.isFinite);
```
Comment (giswqs) [inline apps/geolibre-desktop/src/lib/vector-style-classification.ts]: Fixed in 5fcc1a7c. Supplied numeric values now drop nullish entries, and categorized values retain their original string or number type with type-qualified counting keys. Regression tests cover both cases.

File: apps/geolibre-desktop/src/lib/vector-style-classification.ts [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+  // Any scheme can yield fewer breaks than the requested count when the layer
+  // has few unique values (duplicate breaks collapse); align the color count so
+  // none are dropped.
+  const stopColors =
+    breaks.length === count ? colors : interpolateRampColors(colorRamp, breaks.length);
+
+  return breaks.map((value, index) => ({
+    value: Number(value.toPrecision(8)),
+    color: stopColors[index] ?? stopColors.at(-1) ?? "#2563eb",
+  }));
```
[the remark this answers is not in the document]
Comment (giswqs) [reply apps/geolibre-desktop/src/lib/vector-style-classification.ts:101]: Fixed in 5fcc1a7c. Classification boundaries are stored without precision rounding, with a regression test for adjacent high-magnitude breaks.

File: packages/plugins/src/plugins/maplibre-vector.ts [hunk 4]
```diff
@@ -166,6 +166,20 @@ export async function reloadVectorControlLayer(id: string): Promise<VectorLayerI
...
+ * @param id The store/control layer id.
+ * @param property The attribute field name.
+ * @returns Non-null field values, or null when the control cannot read them.
+ */
+export function getVectorLayerPropertyValues(
+  id: string,
+  property: string,
+): Promise<unknown[] | null> {
+  return vectorControl?.getLayerPropertyValues(id, property) ?? Promise.resolve(null);
+}
```
[the remark this answers is not in the document]
Comment (giswqs) [reply packages/plugins/src/plugins/maplibre-vector.ts]: Fixed in 5fcc1a7c. The wrapper is now async so any synchronous control failure is surfaced as a promise rejection to the caller.

File: apps/geolibre-desktop/src/lib/vector-style-classification.ts [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+  property: string,
+  classCount: number,
+  colorRamp: string,
+  classificationScheme: string,
+  propertyValues?: unknown[],
+): VectorStyleStop[] {
+  const values = (propertyValues ?? getPropertyValues(layer, property))
+    .filter((value) => value !== null && value !== undefined)
+    .map((value) => Number(value))
+    .filter(Number.isFinite);
```
[the remark this answers is not in the document]
Comment (giswqs) [reply apps/geolibre-desktop/src/lib/vector-style-classification.ts]: Fixed in 71eeacfd. Graduated classification now accepts only finite numbers and non-blank numeric strings, with regression coverage for blanks, booleans, arrays, numbers, and numeric strings.

**Comment by giswqs** [comment]
Also fixed the outside-diff attribute availability finding in 71eeacfd. Failed or unavailable reads now remain distinct from successful empty columns, show an inline error, and keep Apply disabled.
